config/v3_0/types/storage_test.go

```diff
@@ -148,3 +148,105 @@ func TestStorageValidate(t *testing.T) {
 		}
 	}
 }
+
+func TestStorageValidateWarnings(t *testing.T) {
+	tests := []struct {
+		in  Storage
+		at  path.ContextPath
+		out error
+	}{
+		{
+			in: Storage{
+				Disks: []Disk{
+					{
+						Device: "/dev/sda",
+						Partitions: []Partition{
+							{}, {},
+						},
+					},
+				},
+				Filesystems: []Filesystem{
+					{
+						Device: "/dev/sda",
+					},
+				},
+			},
+			out: errors.ErrPartitionsOverwritten,
+			at:  path.New("", "filesystems", 0),
+		},
+		{
+			in: Storage{
+				Disks: []Disk{
+					{
+						Device:    "/dev/sda",
+						WipeTable: &[]bool{true}[0],
+					},
+				},
+				Filesystems: []Filesystem{
+					{
+						Device:         "/dev/sda",
+						WipeFilesystem: &[]bool{false}[0],
```

Comment:
I'd expect configs would usually omit this rather than explicitly setting it to false.  We could test both cases, but if we're only testing one, I'd suggest the omitted case.